avahi-python/avahi/__init__.py

```diff
@@ -110,3 +110,22 @@ def dict_to_txt_array(txt_dict):
         l.append(string_to_byte_array("%s=%s" % (k,v)))
 
     return l
+
+
+def txt_array_to_dict(self, txt_array):
+    txt_dict = {}
+    for els in txt_array:
+        key, val = '', None
+        for c in els:
+            c = chr(c)
+            if val is None:
+                if c == '=':
+                    val = ''
+                else:
+                    key += c
+            else:
+                val += c
+        if val is None: # missing '='
+            val = ''
```

Comment:
Is it intentional that this makes it impossible to distinguish between `foo` and `foo=`?

I would have expected the TXT records

```
no_value
empty_value
has_value=foo
```

to come out in Python as, perhaps, something like

```
{
    b'no_value': None,
    b'empty_value': b'',
    b'has_value': b'foo',
}
```

(On one hand, `dict_to_txt_array` doesn't handle that either; on the other hand, perhaps it should, since https://tools.ietf.org/html/rfc6763 does specifically distinguish between an empty value and the absence of a value.)